Src/Base/AMReX_iMultiFab.cpp

```diff
@@ -338,12 +342,14 @@ iMultiFab::sum (int comp, int nghost, bool local) const
             reduce_op.eval(bx, reduce_data,
             [=] AMREX_GPU_DEVICE (int i, int j, int k) -> ReduceTuple
             {
-                return { static_cast<Long>(arr(i,j,k,comp)) };
+                return { static_cast<unsigned long long>(arr(i,j,k,comp))+imax };
```

Comment:
Why not using the macro directly here instead of capturing it?  Also it should be `-INT_MIN` instead because `INT_MIN` is usually `-INT_MAX-1`

Reply:
Are there any warnings/issues/error messages that may interact with type_casting a negative number to an unsigned long long that has a minimum of 0? Or, are the compilers known to be smart enough to do the math in the correct order and never overflow?

This seems to yield an identical time and give the correct answer, baring any overflow concerns:
```               
return { static_cast<unsigned long long>(arr(i,j,k,comp))-
         static_cast<unsigned long long>(INT_MIN) };
```

Note: static_cast<unsigned long long>(-INT_MIN) is outside the int bounds, due to the -INT_MAX-1, so we'd need to do (-(INT_MIN+1)) to suppress the warning, if we wanted to avoid potential issues manually:
```
/ccs/home/kngott/amrex/Src/Base/AMReX_iMultiFab.cpp(346): warning: integer operation result is out of range
```

Reply:
I think it should be `static_cast<unsigned long long>(static_cast<long long>(arr(i,j,k,comp)) - static_cast<long long>(INT_MIN))`.

Reply:
Actually `return { static_cast<unsigned long long>(arr(i,j,k,comp)) - static_cast<unsigned long long>(INT_MIN) };` is fine because it's always well defined with the modulo behavior.